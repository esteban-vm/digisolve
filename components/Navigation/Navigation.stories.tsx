import type { Meta, StoryObj } from '@storybook/react'
import { screen, userEvent } from '@storybook/testing-library'
import Navigation from './Navigation'

export default {
  component: Navigation,
  title: 'Components/Navigation',
  decorators: [
    (Story) => {
      return (
        <div style={{ height: '100vh', width: '100vw', backgroundColor: 'black' }}>
          <Story />
        </div>
      )
    },
  ],
} as Meta<typeof Navigation>

export const NavigationExample: StoryObj<typeof Navigation> = {
  play() {
    const [homeLink, contactLink, aboutLink] = screen.getAllByRole('link')
    const logoImg = screen.getByRole('img')

    userEvent.hover(logoImg)
    userEvent.unhover(logoImg)

    userEvent.click(homeLink)
    userEvent.click(contactLink)
    userEvent.click(aboutLink)
  },
}
