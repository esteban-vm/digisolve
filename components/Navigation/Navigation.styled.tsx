import { styled } from '@/styles'

export const Wrapper = styled.nav`
  & img {
    height: 120px;
    width: auto;
    float: left;
  }
`

export const Menu = styled.ul`
  list-style: none;
  float: right;
  margin-top: 70px;

  & li {
    display: inline-block;
    margin-right: 40px;

    & a {
      &:link,
      &:visited {
        text-decoration: none;
        text-transform: uppercase;
        color: #fff;
        outline: none;
        transition: border-bottom 100ms;
      }

      &:hover,
      &:active {
        border-bottom: 2px solid var(--primary-color);
      }
    }
  }
`
