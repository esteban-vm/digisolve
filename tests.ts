import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
