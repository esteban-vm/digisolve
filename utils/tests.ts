import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
