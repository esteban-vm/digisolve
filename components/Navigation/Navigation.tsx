import type { FC } from 'react'
import Image from 'next/image'
import Link from 'next/link'
import logo from '@/public/img/digisolve-logo.png'
import * as Styled from './Navigation.styled'

const Navigation: FC = () => {
  return (
    <Styled.Wrapper>
      <Image src={logo} alt='Digisolve logo' />
      <Styled.Menu>
        <li>
          <Link href='#'>Home</Link>
        </li>
        <li>
          <Link href='#'>Contact</Link>
        </li>
        <li>
          <Link href='#'>About</Link>
        </li>
      </Styled.Menu>
    </Styled.Wrapper>
  )
}

export default Navigation
