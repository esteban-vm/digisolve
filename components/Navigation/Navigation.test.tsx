import { render, screen } from '@/tests'
import Navigation from './Navigation'

describe('Navigation test cases:', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Navigation />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render nav links', () => {
    render(<Navigation />)

    const links = screen.getAllByRole('link')

    for (const link of links) {
      expect(link).toBeInTheDocument()
      expect(link).toBeVisible()
      expect(link).toHaveAttribute('href', '#')
    }
  })
})
