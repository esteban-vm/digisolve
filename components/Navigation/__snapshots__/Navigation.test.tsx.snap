// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test cases: should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="css-rr22mr"
  >
    <img
      alt="Digisolve logo"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <ul
      class="css-3xog9x"
    >
      <li>
        <a
          href="#"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
